main.tf

```diff
@@ -13,60 +13,43 @@ resource "aws_eip" "default" {
 }
 
 resource "aws_security_group" "default" {
+  #checkov:skip=CKV2_AWS_5:Security group is conditionally attached to Redshift cluster when no existing security groups are provided
+  count = var.subnet_ids != null && length(var.security_group_ids) == 0 ? 1 : 0
+
   name        = "redshift-${var.name}"
   description = "Access to Redshift"
   vpc_id      = var.vpc_id
   tags        = var.tags
+}
 
-  ingress {
```

Comment:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group#removing-all-ingress-and-egress-rules

Reply:
Modified the upgrading doc.